ui/pages/asset/components/asset-page.tsx

```diff
@@ -191,18 +196,26 @@ const AssetPage = ({
           <TokenListItem
             title={symbol}
             tokenSymbol={symbol}
-            primary={`${balance.display} ${symbol}`}
-            secondary={balance.fiat}
+            primary={`${balance} ${symbol}`}
+            secondary={
+              tokenFiatAmount ? formatCurrency(tokenFiatAmount, currency) : ''
+            }
             tokenImage={image}
             isOriginalTokenSymbol={asset.isOriginalNativeSymbol}
             isNativeCurrency={true}
           />
         ) : (
           <TokenCell
+            key={`${symbol}-${address}`}
             address={address}
-            image={image}
+            chainId={chainId}
             symbol={symbol}
-            string={balance.display}
+            image={image}
+            balance={balance}
+            tokenFiatAmount={tokenFiatAmount}
+            string={balance.toString()}
+            decimals={asset.decimals}
+            onClick={() => {}}
```

Comment:
What should this be?

Reply:
Removed this noop, don't think it's required for functionality: https://github.com/MetaMask/metamask-extension/pull/28193/commits/28441c8877967d2d1d999775025a00551ebfec95